lib/ansible/modules/files/copy.py

```diff
@@ -45,14 +45,15 @@
   content:
     version_added: "1.1"
     description:
-      - When used instead of 'src', sets the contents of a file directly to the specified value.
+      - When used instead of I(src), sets the contents of a file directly to the specified value.
         This is for simple values, for anything complex or with formatting please switch to the template module.
     required: false
     default: null
   dest:
     description:
-      - Remote absolute path where the file should be copied to. If src is a directory,
-        this must be a directory too.
+      - Remote absolute path where the file should be copied to. If I(src) is a directory, this must be a directory too and
+        I(dest) directory is created when it doesn't exist.
+        If I(src) and I(dest) are files, then parent directory of I(dest) isn't created: the task fails if it doesn't already exist.
     required: true
```

Comment:
So this is now incomplete in a different way.

Parent directories of `dest` are created automatically, if 1) `dest` ends in `/` OR 2) `src` is a directory.

The previous commit only mentioned case 1).  This new version only mentions case 2).

I wrote a line in my RFE for `dest_dir`, #25717.  I think fully documenting the behaviour of `dest` will not remove all confusion, because the behaviour is inherently confusing.  The `dest_dir` RFE is my best attempt, to provide a less confusing behaviour, and save people from having to think about what `dest` does.